src/Orleans/Messaging/GatewayClientReceiver.cs

```diff
@@ -231,10 +231,7 @@ private int FillBuffer(List<ArraySegment<byte>> buffer)
                 // Only try to reconnect if we're not shutting down
                 if (Cts.IsCancellationRequested) return 0;
 
-                if (ex is SocketException)
-                {
-                    Log.Warn(ErrorCode.Runtime_Error_100158, "Exception receiving from gateway " + gatewayConnection.Address);
-                }
+                Log.Warn(ErrorCode.Runtime_Error_100158, String.Format("Exception receiving from gateway {0}: {1}", gatewayConnection.Address, ex.Message));
```

Comment:
Shouldn't we remove the commented out batching code above `FillBuffer()`? That code is dead now that we changed the socket read side which made it incompatible with this old approach to batching. Since you are touching the source file, I would use that as an opportunity to clean it up.